spec/features/saml/loa3_sso_spec.rb

```diff
@@ -33,6 +33,22 @@
     end
   end
 
+  context 'canceling verification', js: true do
+    it 'returns user to profile page if they have previously signed up' do
+      sign_in_and_2fa_user
+
+      loa3_sp_session do |session|
+        session['warden.user.user.session'][:first_time_recovery_code_view] = true
+      end
+
+      visit verify_path
+
+      click_on t('links.cancel')
+      click_on t('loa3.buttons.cancel')
+      expect(current_url).to match(/profile/)
```

Comment:
I'm not sure this test reflects reality, and I think it's masking a bug. If you remove `session['warden.user.user.session'][:first_time_recovery_code_view] = true` from this test, it will simulate a user who has finished creating their account, then makes an LOA3 request, and tries to cancel the verification. I would expect that they be sent directly to the profile screen. Instead, they are presented with the recovery code confirmation. I don't think already-signed-up users who are canceling verification need to view the recovery code screen, right?

However, if a user is not already signed up, and makes an LOA3 request, then in that case they do need to see the recovery code screen after canceling because in that scenario, we only show the recovery code screen after IdV, not during the regular account creation process (to avoid doing it twice).